actionview/lib/action_view/helpers/asset_tag_helper.rb

```diff
@@ -77,12 +80,20 @@ module AssetTagHelper
       def javascript_include_tag(*sources)
         options = sources.extract_options!.stringify_keys
         path_options = options.extract!("protocol", "extname", "host", "skip_pipeline").symbolize_keys
-        sources.uniq.map { |source|
+        early_hints_links = []
+
+        sources_tags = sources.uniq.map { |source|
+          href = path_to_javascript(source, path_options)
+          early_hints_links << { link: href, as: "script" }
           tag_options = {
-            "src" => path_to_javascript(source, path_options)
+            "src" => href
           }.merge!(options)
           content_tag("script".freeze, "", tag_options)
         }.join("\n").html_safe
+
+        request.send_early_hints(early_hints_links)
```

Comment:
This reads to me as if users can only call `javascript_include_tag` once, is that right?

It's possible people have calls in their application layout and then in individual template files (I'm not saying that's a good idea, but I think that does exist).

Reply:
I wrote it this way so that if you pass multiple sources into `javascript_include_tag` that it will only send early hints once, but you can call `javascript_include_tag` as many times as you want and it will send early hints.